core/src/cluster_info_repair_listener.rs

```diff
@@ -305,20 +305,30 @@ impl ClusterInfoRepairListener {
                         // a database iterator over the slots because by the time this node is
                         // sending the blobs in this slot for repair, we expect these slots
                         // to be full.
-                        if let Some(blob_data) = blocktree
-                            .get_data_blob_bytes(slot, blob_index as u64)
+
+                        if let Some(data_blob) = blocktree
+                            .get_data_blob(slot, blob_index as u64)
                             .expect("Failed to read data blob from blocktree")
                         {
-                            socket.send_to(&blob_data[..], repairee_tvu)?;
+                            socket.send_to(&data_blob.data[..], repairee_tvu)?;
                             total_data_blobs_sent += 1;
-                        }
 
-                        if let Some(coding_bytes) = blocktree
-                            .get_coding_blob_bytes(slot, blob_index as u64)
-                            .expect("Failed to read coding blob from blocktree")
-                        {
-                            socket.send_to(&coding_bytes[..], repairee_tvu)?;
-                            total_coding_blobs_sent += 1;
+                            if let Some(coding_header) = data_blob.get_coding_header() {
+                                for coding_blob_index in 0..coding_header.parity_count as u64 {
```

Comment:
we send *all* the coding blobs surrounding a data_blob in this case?

Reply:
The only alternative I can think of is to somehow communicate how many coding blobs the other peer already has

Reply:
just to be clear: if we're doing 8 data + 8 coding, this repair response sends 9 blobs back?

how about picking one at random from the same set?

Reply:
the erasure set has a coding/data ratio > 1, send that ratio